### ISSUE TITLE
Network tab

### ISSUE THREAD

**Issue by makslevental** [GoogleChromeLabs/ndb#12]
https://news.ycombinator.com/item?id=17581719

this is a thing i've always wanted that i've never known i've wanted until now. seems like it should be possible given that this is dev tools ripped out of chrome but of course i'm not sure given how massively complex this project is. just wondering.

thanks for the work!

**Comment by ebidel** [comment]
@patrickhulce I would say both? A node server is a good use case where you want to see incoming requests to your server and outgoing requests made by the app/backend. Points for being able to filter between incoming/outgoing requests.

**Comment by bensleveritt** [comment]
Literally first thing I looked for after seeing the post on HN.

@patrickhulce Both if possible, it’d be a complete life saver to see the requests as they fly in and out, rather than debugging just the payload before/after flight, or proxying the requests through another app.

**Comment by patrickhulce** [comment]
Yeah makes sense I've just been thinking lately that it'll require a completely new UX. I'd love to see incoming routes with a breakdown of time spent in each middleware, incoming request, final response body, etc. I could see the existing network panel being extremely confusing if outgoing and incoming were intermingled (it's already bad enough when you're trying to look at a service worker page 😆)

**Comment by nyxtom** [comment]
If it's easier to just add in outgoing requests that would be really helpful.

**Comment by ak239** [comment]
We are working on this feature request. Stay tuned.

**Comment by khanghoang** [comment]
@ak239 I'm interested in this feature, is there a way I can help?

**Comment by khanghoang** [comment]
I was hacking around to make it work. Stay tuned!!!!

![ndb 2018-12-17 10-52-54](https://user-images.githubusercontent.com/3213579/50108385-ea372100-01e9-11e9-82bf-b0998752557b.png)

**Comment by carylewis** [comment]
This would be a fantastic feature!

**Comment by a7madgamal** [comment]
any workarounds for getting this will be amazing

**Comment by khanghoang** [comment]
@a7madgamal WIP fork https://github.com/khanghoang/ndb#master

**Comment by khanghoang** [comment]
I'm thrilling to announce that I'm working on WIP PR for this #282

**Comment by GrinZero** [comment]
想知道进度。。。
